### PR TITLE
Internal: Enable different top section content for Free/Pro Users on the Home screen [ED-14519]

### PR DESCRIPTION
**Pull request by elchugreeva**

### CODE REVIEW

File: modules/home/transformations/filter-top-section-by-license.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class Filter_Top_Section_By_License extends Transformations_Abstract {
+	public bool $has_pro;
+
+	public function __construct( $args ) {
+		parent::__construct( $args );
+
+		$this->has_pro = Utils::has_pro();
+	}
+
+	private function valid_item( $item ) {
```
Comment (TzviRabinovitch) [inline modules/home/transformations/filter-top-section-by-license.php:19]: Maybe is_valid_item ?

File: modules/home/transformations/filter-top-section-by-license.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		if ( isset( $item['license'] ) ) {
+			$has_pro_json_not_free = $this->has_pro && 'pro' === $item['license'][0];
+			$is_not_pro_json_not_pro = ! $this->has_pro && 'free' === $item['license'][0];
+
+			return $has_pro_json_not_free || $is_not_pro_json_not_pro;
+		}
+	}
+
+	public function transform( array $home_screen_data ): array {
+		if ( is_array( $home_screen_data['top'] ) && count( $home_screen_data['top'] ) > 1 && is_array( $home_screen_data['top'][0] ) ) {
```
Comment (TzviRabinovitch) [inline modules/home/transformations/filter-top-section-by-license.php:29]: Can you reduce nesting ?

Comment (TzviRabinovitch) [inline modules/home/transformations/filter-top-section-by-license.php:29]: Maybe extract this condition to a function that explains the check ?

File: modules/home/transformations/filter-top-section-by-license.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+			$new_top = [];
+
+			foreach ( $home_screen_data['top'] as $index => $item ) {
+				if ( $this->valid_item( $item ) ) {
```
Comment (TzviRabinovitch) [inline modules/home/transformations/filter-top-section-by-license.php:33]: Maybe extract this to a function that dies the check and pushes to the array if the check passed

File: modules/home/transformations/filter-top-section-by-license.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	public function transform( array $home_screen_data ): array {
+		foreach ( $home_screen_data['top_with_licences'] as $index => $item ) {
+			if ( $this->is_valid_item( $item ) ) {
+				$new_top[] = $item;
+			}
+		}
+
+			$home_screen_data['top_with_licences'] = reset( $new_top );
```
Comment (hein-obox) [inline modules/home/transformations/filter-top-section-by-license.php:35]:
```suggestion
			$home_screen_data['top'] = reset( $new_top );
```

File: modules/home/transformations/filter-top-section-by-license.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+			return $home_screen_data;
```
Comment (hein-obox) [inline modules/home/transformations/filter-top-section-by-license.php:37]:
```suggestion
		return $home_screen_data;
```

File: modules/home/transformations/filter-top-section-by-license.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+				$new_top[] = $item;
+			}
+		}
+
+		$home_screen_data['top_with_licences'] = reset( $new_top );
+		unset( $home_screen_data['top'] );
+
+		return $home_screen_data;
+
+	}
```
Comment (hein-obox) [inline modules/home/transformations/filter-top-section-by-license.php:40]: Remove empty line below the `return`.

**Review by hein-obox — APPROVED** [review]

**Pull request merged**